src/shogun/features/Features.cpp

```diff
@@ -28,10 +28,15 @@ CFeatures::CFeatures(const CFeatures& orig)
 {
 	init();
 
-	// Call to init creates new preproc arrays.
-	SG_UNREF(preproc);
-	preproc = orig.preproc;
-	SG_REF(preproc);
+	// TODO this should be a shallow copy
+	auto old_preproc = preproc;
+	preproc = make_clone(preproc);
+	SG_UNREF(old_preproc);
+
+	// TODO this should be a shallow copy
```

Comment:
@lisitsyn this is what I need the shallow clone method for (just recursively create new instances, but always share the data (vectors/matrices, features, etc)